easybuild/easyconfigs/r/R-bundle-Bioconductor/R-bundle-Bioconductor-3.9-foss-2019a-R-3.6.0.eb

```diff
@@ -31,7 +31,12 @@ exts_default_options = {
 }
 
 exts_defaultclass = 'RPackage'
-exts_filter = ("R -q --no-save", "library(%(ext_name)s)")
+
+# check whether correct version is installed in extension filter
+# (some versions in this bundle may be newer than the ones provided by R)
+local_ext_version_check = "if (packageVersion('%(ext_name)s') != '%(ext_version)s')"
+local_stop_msg = "stop('%(ext_name)s %(ext_version)s not installed')"
```

Comment:
@boegel maybe can we differentiate the message here?
between packages that have not been installed and packages that have been installed but with the wrong version

Reply:
@smoors That's already the case. If a package is not installed, `packageVersion` will fail, and the `stop` will not be run.

Basically, this:

```
# non-existing package
$ echo "if (packageVersion('foo') != '1.2.3') stop('foo 1.2.3 not installed')" | R -q --no-save
> if (packageVersion('foo') != '1.2.3') stop('foo 1.2.3 not installed')
Error in packageVersion("foo") : there is no package called ‘foo’
Execution halted
$ echo $?
1
```

```
# installed package but different version
$ echo "if (packageVersion('R.utils') != '1.2.3') stop('R.utils 1.2.3 not installed')" | R -q --no-save
> if (packageVersion('R.utils') != '1.2.3') stop('R.utils 1.2.3 not installed')
Error: R.utils 1.2.3 not installed
Execution halted
$ echo $?
1
```

Reply:
Hmm, perhaps this is better though:
```
$ echo "pkgver = packageVersion('R.utils'); if (pkgver != '1.2.3') stop('R.utils 1.2.3 not installed, found ', pkgver)" | R -q --no-save
> pkgver = packageVersion('R.utils'); if (pkgver != '1.2.3') stop('R.utils 1.2.3 not installed, found ', pkgver)
Error: R.utils 1.2.3 not installed, found 2.8.0
Execution halted
```

fixed in 8b91562